Source/WebKit/NetworkProcess/storage/SQLiteStorageArea.cpp

```diff
@@ -405,5 +442,15 @@ void SQLiteStorageArea::handleLowMemoryWarning()
         m_database->releaseMemory();
 }
 
+void SQLiteStorageArea::handleDatabaseCorruption()
```

Comment:
Since we are checking the error in all the places, maybe we can change this to something like:
```
bool SQLiteStorageArea::handleDatabaseCorruptionIfNeeded(int databaseError)
{
    if (databaseError != SQLITE_CORRUPT && databaseError != SQLITE_NOTADB)
        return false;

    m_database = nullptr;
    RELEASE_LOG(Storage, "SQLiteStorageArea::handleDatabaseCorruption deletes corrupted database file '%s'", m_path.utf8().data());
    webcore::SQLiteFileSystem::deleteDatabaseFile(m_path);
    return true;
}
```
and use handleDatabaseCorruptionIfNeeded(result) in all the places.